### PR TITLE
Add typing in GeneralizedRCNNTransform

### PR DESCRIPTION
**Pull request by datumbox**

### CODE REVIEW

File: torchvision/models/detection/transform.py [hunk 1]
```diff
@@ -86,10 +85,9 @@ def __init__(self, min_size, max_size, image_mean, image_std, size_divisible=32,
         self.fixed_size = fixed_size
 
     def forward(self,
-                images,       # type: List[Tensor]
-                targets=None  # type: Optional[List[Dict[str, Tensor]]]
-                ):
-        # type: (...) -> Tuple[ImageList, Optional[List[Dict[str, Tensor]]]]
+                images: List[Tensor],
+                targets: Optional[List[Dict[str, Tensor]]] = None
+                ) -> Tuple[ImageList, Optional[List[Dict[str, Tensor]]]]:
```
Comment (oke-aditya) [inline torchvision/models/detection/transform.py:90]: Hey @datumbox !
We don't need to hack this anyomore!
Unoin is supported by Torchscript #4360 

We can use `Union[ImageList, Optional[List[Dict[str, Tensor]]]`

I'm currently working to use this and possibly finish #4220

Comment (oke-aditya) [reply torchvision/models/detection/transform.py:90]: Aah found it https://github.com/pytorch/vision/pull/4220#discussion_r679218040 yeah 
I commented on wrong code.

Comment (datumbox) [reply torchvision/models/detection/transform.py:90]: I'm not sure this is related to the problem you describe. This is not a workaround, could you clarify if your comment is still valid here?

Comment (oke-aditya) [reply torchvision/models/detection/transform.py:90]: Yeah its' invalid. My bad

Comment (datumbox) [reply torchvision/models/detection/transform.py:90]: no worries :) thanks for checking though

**Review by fmassa — APPROVED** [review]
Thanks!

**Pull request merged**